static/app/views/organizationGroupDetails/groupSimilarIssues/similarStackTrace/index.tsx

```diff
@@ -36,83 +37,31 @@ type State = {
   v2: boolean;
 };
 
-class SimilarStackTrace extends Component<Props, State> {
-  state: State = {
+function SimilarStackTrace({params, location, project}: Props) {
+  const {orgId, groupId} = params;
+
+  const [state, setState] = useState<State>({
```

Comment:
I remember we discussed this during a [Front End TSC meeting](https://www.notion.so/sentry/2022-06-02-8483342f5ba649f29c18a635efd084e1) and later in a Slack thread that unfortunately we no longer have.

If I'm not mistaken, the conclusion was that we should only use many items in a single object if we are using a reducer to manage the component state, otherwise, it is better if they are separated. You can find some examples with `reducer` in the profiling code cc @JonasBa

Reply:
Actually that was a bit unrelated. The idea behind that state is that you dont have a primitive value state that you nest under an unnecessary object. For example, `[{count}, setCount] = useState({count: 0})` should be `const [count, setCount] =. useState(0)`.

In cases where you have interconnected state, it is probably better to keep it all in a single state object, else tracking all the effects and values becomes tedious (but you do have the benefit of slightly better optimized state updates).